extmod/modbluetooth_nimble.c

```diff
@@ -614,16 +614,24 @@ STATIC int gap_scan_cb(struct ble_gap_event *event, void *arg) {
     return 0;
 }
 
-int mp_bluetooth_gap_scan_start(int32_t duration_ms) {
+int mp_bluetooth_gap_scan_start(int32_t duration_ms, int32_t interval_ms, int32_t window_ms) {
     if (duration_ms == 0) {
         duration_ms = BLE_HS_FOREVER;
     }
-    STATIC const struct ble_gap_disc_params discover_params = {
-        .itvl = BLE_GAP_SCAN_SLOW_INTERVAL1,
-        .window = BLE_GAP_SCAN_SLOW_WINDOW1,
+    uint16_t interval = BLE_GAP_SCAN_SLOW_INTERVAL1;
+    uint16_t window = BLE_GAP_SCAN_SLOW_WINDOW1;
+    if (interval_ms > 0) {
```

Comment:
No need to check this, just assume valid (default at least) values are passed in.